zbus/src/address/transport/unixexec.rs

```diff
@@ -0,0 +1,86 @@
+use std::{borrow::Cow, ffi::OsStr, fmt};
+
+use super::{
+    percent::{decode_percents_os_str, decode_percents_str, EncOsStr},
+    Address, Error, KeyValFmt, KeyValFmtAdd, Result,
+};
+
+#[derive(Debug, PartialEq, Eq)]
+struct Argv(usize);
+
+impl fmt::Display for Argv {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let n = self.0;
+
+        write!(f, "argv{n}")
+    }
+}
+
+/// `unixexec:` D-Bus transport.
+///
+/// <https://dbus.freedesktop.org/doc/dbus-specification.html#transports-exec>
+#[derive(Debug, PartialEq, Eq)]
+pub struct Unixexec<'a> {
+    path: Cow<'a, OsStr>,
+    argv: Vec<(usize, Cow<'a, str>)>,
```

Comment:
why keep the index in a Vec? Vec and slices are indexed already.

Reply:
First, because you may or may not have argv0... 

You may also have multiple argvN (since we don't enforce this anymore)

And also the spec says " If a specific argvX is not specified no further argvY for Y > X are taken into account." which may be interepreted as the address may be accepted with argvX missing but further argvY.. in which case you should ignore them for exec, but you may still want to retain them?..

Reply:
Unresolving this thread for this part:

> > First, because you may or may not have argv0...
> 
> That just means that argv0 needs special treatment (sperate field/getter).

This is also closer to the spec since arg0 is separate key.